hadoop-tools/hadoop-aws/src/main/java/org/apache/hadoop/fs/s3a/Constants.java

```diff
@@ -429,6 +429,24 @@ private Constants() {
    */
   public static final String CONTENT_ENCODING = "fs.s3a.object.content.encoding";
 
+  /**
+   * S3 storage class: standard, reduced_redundancy, intelligent_tiering etc.
+   */
+  public static final String STORAGE_CLASS = "fs.s3a.storage.class";
+
+  /**
+   * All possible values for storage class.
+   */
+  public static final String STORAGE_CLASS_STANDARD = "standard";
+  public static final String STORAGE_CLASS_REDUCED_REDUNDANCY = "reduced_redundancy";
+  public static final String STORAGE_CLASS_GLACIER = "glacier";
+  public static final String STORAGE_CLASS_STANDARD_INFREQUENT_ACCESS = "standard_ia";
+  public static final String STORAGE_CLASS_ONEZONE_INFREQUENT_ACCESS = "onezone_ia";
+  public static final String STORAGE_CLASS_INTELLIGENT_TIERING = "intelligent_tiering";
+  public static final String STORAGE_CLASS_DEEP_ARCHIVE = "deep_archive";
+  public static final String STORAGE_CLASS_OUTPOSTS = "outposts";
+  public static final String STORAGE_CLASS_GLACIER_INSTANT_RETRIEVAL = "glacier_ir";
+
```

Comment:
We will have to keep updating these values. However, users will not be effected by this because these constants are used in our code only, users can still put any value they like in xml file.

also discussed with Steve, he prefer to have static string in Constants.